protocol/communities_messenger_token_permissions_test.go

```diff
@@ -445,6 +445,110 @@ func (s *MessengerCommunitiesTokenPermissionsSuite) TestJoinedCommunityMembersSe
 	}
 }
 
+func (s *MessengerCommunitiesTokenPermissionsSuite) TestEditSharedAddresses() {
+	community, _ := s.createCommunity()
+	s.advertiseCommunityTo(community, s.alice)
+
+	s.joinCommunity(community, s.alice, alicePassword, []string{aliceAddress2})
+
+	community, err := s.owner.GetCommunityByID(community.ID())
+	s.Require().NoError(err)
+
+	s.Require().Equal(2, community.MembersCount())
+
+	for pubKey, member := range community.Members() {
+		if pubKey != common.PubkeyToHex(&s.owner.identity.PublicKey) {
+			s.Require().Len(member.RevealedAccounts, 1)
+
+			switch pubKey {
+			case common.PubkeyToHex(&s.alice.identity.PublicKey):
+				s.Require().Equal(member.RevealedAccounts[0].Address, aliceAddress2)
+			default:
+				s.Require().Fail("pubKey does not match expected keys")
+			}
+		}
+	}
+
+	passwdHash := types.EncodeHex(crypto.Keccak256([]byte(alicePassword)))
+	request := &requests.EditSharedAddresses{CommunityID: community.ID(), Password: passwdHash, AddressesToReveal: []string{aliceAddress1}}
+	response, err := s.alice.EditSharedAddressesForCommunity(request)
+	s.Require().NoError(err)
+	s.Require().NotNil(response)
+
+	// Retrieve address change
+	err = tt.RetryWithBackOff(func() error {
+		response, err := s.owner.RetrieveAll()
+		if err != nil {
+			return err
+		}
+		if len(response.Communities()) == 0 {
+			return errors.New("no communities in response (address change reception)")
+		}
+		community := response.Communities()[0]
+		for pubKey, member := range community.Members() {
+			if pubKey != common.PubkeyToHex(&s.owner.identity.PublicKey) {
+				switch pubKey {
+				case common.PubkeyToHex(&s.alice.identity.PublicKey):
+					if member.RevealedAccounts[0].Address != aliceAddress1 {
+						return errors.New("address not yet updated")
+					}
+				default:
+					s.Require().Fail("pubKey does not match expected keys")
+				}
+			}
+		}
+		return nil
```

Comment:
minor: this code repeated 3 times, can be moved to lambda and reused where we need it